### DISCUSSION TITLE
How to deal with multiple select elements on one form and livewire page

### DISCUSSION THREAD

**Discussion by stojankukrika** [livewire/livewire#4145] [1. Help] [answered]
Hi,
Can you help me with how to deal with multiple select elements on one form and livewire page? I tried but when I change one and in a method like 'updatedDate' which is called after date changed the other two parameters 'city_from_id' and 'city_to_id' and others are null. How to deal with it? I need to make a form that will check available routes depending on two city params and one date parameter. And if all Ok then the user can fill up the rest of the form. But I can't find a way how to make it.

Date and cityFrom and cityTo are not dependent on each other, so users will get info when selecting all data or on changing some info another select box (CityFrom and CityTo) can be changed (profiled aveable values).

I put the question here also:
https://stackoverflow.com/questions/69814526/how-to-deal-with-multiple-select-elements-on-one-form-and-livewire-page

**Comment by joshhanley** [comment]
@stojankukrika what you can do is create a computed property, that returns true if all 3 fields have been populated and false if not.

Then in your blade view, you can use the computed property to determine whether to disable the other fields or not.

See here for details https://laravel-livewire.com/docs/2.x/properties#computed-properties

Hope this helps!

**Reply by stojankukrika** [reply]
@joshhanley Can you give me example how to make it? I read doc but it's not clear to me how to make it? For code example you can take a look here
https://stackoverflow.com/questions/69814526/how-to-deal-with-multiple-select-elements-on-one-form-and-livewire-page

**Comment by joshhanley** [comment]
@stojankukrika sure!

You would setup your computed property like this, so this would be for a computed property called `$this->flightIsAvailable` which returns true or false depending on if it is available or not:

```php
public function getFlightIsAvailableProperty()
{
    if ($this->search['date'] == null || $this->search['city_from_id']  == null || $this->search['city_to_id'] == null) {
        return false;
    }
    
    // Run your check here to see if the flight is available
    $isAvailable = FlightService::check($this->search['date'], $this->search['city_from_id'], $this->search['city_to_id']);
    
    if($isAvailable) {
        return true;
    }
    
    // If flight is not available you could throw a validation error or browser event or something here so the user knows it's not available.
    return false;    
}
```

Then on any inputs that you want disabled when the flight is not available, like this customer name input, you would do something like this:

```blade
<input type="text" :disabled="! $this->flightIsAvailable" wire:model="customerName" />
```

Hope this helps!

**Reply by stojankukrika** [reply]
Maybe we didn't understand each other. If I understand this code well, this will disable/enable enter customerName in form?
But I need some way to bind all three values from $search array ('search['city_from_id'], search['date'] and search['city_to_id']). I have all three params in form. When I change one I have only that one, loose prev. selected values for others and always have only one value in $search property, and on the blade file(which is render - show to the user, values from prev. selected values are shown as selected but in Component code, I only have the last one), in that way, I can't find anything because the need to have all three to make some search and give customer results.
Is there any option to store the previously selected value? Please take a look attached video.

https://user-images.githubusercontent.com/10199584/140253084-2d6879d2-1ed6-409e-8e59-a4da495e45d9.mov

**Comment by joshhanley** [comment]
@stojankukrika you said this in your original post:

> I need to make a form that will check available routes depending on two city params and one date parameter. And if all Ok then the user can fill up the rest of the form. But I can't find a way how to make it.

That is what I gave you a solution for as I thought that request was pretty clear 🙂

In regards to the missing data, it's because you are using `dd` which means dump and **die**. So the rest of the Livewire request isn't finishing, hence the data isn't stored correctly in Livewire's state.

Try changing the `dd()` to `dump()` 🙂

Hope this helps!

**Reply by stojankukrika** [reply]
I get the same :( with dd and dump if I dump in this way:
` dump($this->search['date']??'DATE');
   dump($this->search['city_from_id']??'CITY FROM');
  dump($this->search['city_to_id']??'CITY TO');`
  just get 'DATE' 3 'CITY TO', or some other combination but can't get all three at once.

**Comment by joshhanley** [comment, marked as answer]
@stojankukrika hmm ok, can you please share your full component class and blade view here?

**Reply by stojankukrika** [reply]
Sure @joshhanley , here it is https://gist.github.com/stojankukrika/de0d19d5f19b8006f0a41d9e5d8df463 it's is my last version of this code

**Reply by stojankukrika** [reply]
@joshhanley I findout how it works... without dd or dump it works.... but why? why does not work with dd or dump :( ?
